src/models/thread.ts

```diff
@@ -377,20 +378,19 @@ export class Thread extends ReadReceipt<ThreadEmittedEvents, ThreadEventHandlerM
             // When there's no server-side support, just add it to the end of the timeline.
             this.addEventToTimeline(event, toStartOfTimeline);
             this.client.decryptEventIfNeeded(event);
-        } else if (!toStartOfTimeline && this.initialEventsFetched && isNewestReply) {
-            // When we've asked for the event to be added to the end, and we're
-            // not in the initial state, and this event belongs at the end, add it.
-            this.addEventToTimeline(event, false);
-            this.fetchEditsWhereNeeded(event);
         } else if (event.isRelation(RelationType.Annotation) || event.isRelation(RelationType.Replace)) {
             this.addRelatedThreadEvent(event, toStartOfTimeline);
             return;
-        } else if (this.initialEventsFetched) {
-            // If initial events have not been fetched, we are OK to throw away
-            // this event, because we are about to fetch all the events for this
-            // thread from the server.
-
+        } else if (!toStartOfTimeline && isNewestReply) {
+            // When we've asked for the event to be added to the end,
+            // and this event belongs at the end, add it.
+            this.addEventToTimeline(event, false);
+            this.fetchEditsWhereNeeded(event);
+        } else {
             // Otherwise, we should add it, but we suspect it is out of order.
+            // This may be because we've just created the thread object and are
+            // still fetching events, in which case add it where we think is sensible
+            // and it will be replaced later once the pagination request completes.
```

Comment:
> and it will be replaced later

how will it be replaced? don't we just addEvent each of the events returned by the pagination request? We don't remove any existing events?

Otherwise what is the point of doing `resetLiveTimeline` before the pagination request?

Reply:
I'm not sure I understand. `resetLiveTimeline` removes the events that get added here before they get replaced by the result of the pagination request. Does c45def0e2 make it any clearer?

Reply:
I guess my question really amounts to: is there any way we can hit this codepath while `paginateEventTimeline` is running?

Something like:

 * we start loading the initial events, beginning with `resetLivetTimeline` then the start of a call to `paginateEventTimeline`
 * an event arrives over /sync which is added above
 * we back-paginate from the new event, resulting in new events which hit this codepath
 * the initial event load finally completes.

According to this comment, the events added in step 3 are removed, but that doesn't seem correct?

Reply:
Ah right yes, I think I see what you mean: really we would want to call `resetLivetTimeline` after the async bit complete but before the events from the pagination request start to be added, but `paginateEventTimeline` doesn't allow for that: we'd have to split out something separate to allow us to do that. I think I'm inclined to comment this and fix it separately: I don't think it's a trivial thing to fix as part of this PR. Wdyt?

Reply:
Fwiw I have commented this in f0dc6014a